src/common/src/logger.cpp

```diff
@@ -1,7 +1,19 @@
 #include "novatel_edie/common/logger.hpp"
 
-LoggerManager::~LoggerManager() = default;
-
 std::unique_ptr<LoggerManager> pclLoggerManager = std::make_unique<CPPLoggerManager>();
 
+LoggerManager::~LoggerManager() = default;
+
 CPPLoggerManager* GetLoggerManager() { return dynamic_cast<CPPLoggerManager*>(pclLoggerManager.get()); }
+
+LoggerManager* GetBaseLoggerManager()
+{
+    if (pclLoggerManager == nullptr) { throw std::runtime_error("GetBaseLoggerManager(): LoggerManager is not initialized"); }
+    return pclLoggerManager.get();
```

Comment:
I would recommend initializing `pclLoggerManager` lazily, since initialization of global static is somewhat fragile: https://en.cppreference.com/w/cpp/language/siof.html
The standard guarantees that static initialization is handled in a thread-safe manner: https://timsong-cpp.github.io/cppwp/n3337/stmt.dcl#4
```suggestion
    if (pclLoggerManager == nullptr) { pclLoggerManager = std::make_unique<CPPLoggerManager>(); }
    return pclLoggerManager.get();
```